spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/config/KafkaBinderConfiguration.java

```diff
@@ -151,14 +154,20 @@ public boolean matches(ConditionContext conditionContext, AnnotatedTypeMetadata
 		public boolean matches(ConditionContext conditionContext, AnnotatedTypeMetadata annotatedTypeMetadata) {
 
 			ClassLoader classLoader = Kafka09Condition.class.getClassLoader();
-			Method addPartitions = getMethod(classLoader, "addPartitions");
-			if (addPartitions != null) {
-				Class<?>[] parameterTypes = addPartitions.getParameterTypes();
-				Class clazz = parameterTypes[parameterTypes.length - 1];
-				if (!clazz.getName().equals("kafka.admin.RackAwareMode")) {
-					return true;
+			try {
```

Comment:
One suggestion, missed when merged - but can we use `org.apache.kafka.common.utils.AppInfoParser#getVersion` to detect the Kafka client library version instead of the signature.